### ISSUE TITLE
Emacs 29 renamed native-comp-deferred-compilation-deny-list

### ISSUE THREAD

**Issue by snf** [radian-software/straight.el#1059]
Emacs 29 renamed `native-comp-deferred-compilation-deny-list` to `native-comp-jit-compilation-deny-list` so `straight--build-native-compile` breaks.

I can open a PR fixing this (tested in my local build) but I don't know what's the best way to maintain backwards compatibility with the older symbol.

Reference: https://github.com/emacs-mirror/emacs/commit/95692f6754c3a8f55a90df2d6f7ce62be55cdcfc

**Comment by flintforge** [comment]
What about this ?
in [[file:~/.emacs.d/straight/repos/straight.el/straight.el::defun straight--build-native-compile]]

```elisp
(defvar native-comp-jit-compilation-deny-list
  (if (< (string-to-number emacs-version) 29)
      native-comp-deferred-compilation-deny-list
      native-comp-jit-compilation-deny-list
      ))


```
and replace `native-comp-deferred-compilation-deny-list`
down below by `native-comp-jit-compilation-deny-list`

**Comment by snehrbass** [comment]
being slightly pedantic, the exact warning I am getting is 
```
straight.el:5236:33: Warning: ‘native-comp-deferred-compilation-deny-list’ is an obsolete variable (as of 29.1);
```
not sure if supporting the one minor release is that important.

**Comment by progfolio** [comment]
Was this not fixed by ad1b02e on the "develop" branch?

**Comment by johnhamelink** [comment]
> Was this not fixed by [ad1b02e](https://github.com/radian-software/straight.el/commit/ad1b02e921a28c3cb532202a74a87e1cc2156810) on the "develop" branch?

This does fix it.

I was also able to alias the variable while marking it deprecated like so:

```elisp
(define-obsolete-variable-alias
  'native-comp-deferred-compilation-deny-list
  'native-comp-jit-compilation-deny-list
  "Renamed in emacs#95692f6")
```

**Comment by snf** [comment]
Thank you, missed the fix because assumed this was bootstrapping from the develop branch.

After RTFMing, this actually bootstraps from the develop branch that just works with Emacs 29: `(setq straight-repository-branch "develop")`
